tests/spec/localhost/nvidia_spec.rb

```diff
@@ -0,0 +1,13 @@
+require 'spec_helper'
+
+has_nvidiactl = file('/dev/nvidiactl').exists?
+
+# should always be installed
+describe file('/usr/bin/nvidia-docker') do
+  it { should exist }
+end
+
+# we just can test this on gpu enabled instances.
+describe command('nvidia-docker run --rm nvidia/cuda nvidia-smi'), :if => has_nvidiactl do
+  its(:stdout) { should contain /Tesla K80/ }
```

Comment:
@elezar you're right nvidia-docker still runs without gpus. So we should be able to test. But if you try the usual `docker run --rm nvidia/cuda nvidia-smi` it ends up in a not found...  However IMO it should be enough to have a test for the nvidia-docker executable + having a conditional check on GPU instances.